homeassistant/components/google/__init__.py

```diff
@@ -322,11 +329,8 @@ def load_config(path: str) -> dict[str, Any]:
         with open(path, encoding="utf8") as file:
             data = yaml.safe_load(file)
             for calendar in data:
-                try:
-                    calendars.update({calendar[CONF_CAL_ID]: DEVICE_SCHEMA(calendar)})
-                except VoluptuousError as exception:
-                    # keep going
-                    _LOGGER.warning("Calendar Invalid Data: %s", exception)
+                validated = DEVICE_SCHEMA(calendar)
+                calendars.update({validated[CONF_CAL_ID]: validated})
```

Comment:
```suggestion
                calendars[validated[CONF_CAL_ID]] =  DEVICE_SCHEMA(calendar)
```

Reply:
1) I don't think this suggestion works given `validated` is not a variable anymore.
2) Assuming that is rewritten as `calendar`, then part of the original bug was it throws `KeyError` because it assumed that `CONF_CAL_ID` is present in the dict. This is why its split into these two ugly verbose steps.

Reply:
My bad, updated now to rename `validated` to `calendar`. The reason why I think it should work is because it first evaluates the right hand, so by that time we can be certain that `calendar[CONF_CAL_ID]` resolves? 

Anyway, we don't have to do it, I mainly started the suggestion to rewrite the use from `.update` to using a direct key assignment.

Reply:
Can you help me understand, then, why the stacktrace in the PR happens? I think this is what was here before.

Reply:
hm I'll be damned, you're right it is evaluating the key first.

```
>>> a = {}
>>> b = {}
>>> a.update({b["nope"]: int()})
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
KeyError: 'nope'
>>> a[b["nope"]] = int()
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
KeyError: 'nope'
```

Reply:
nevermind, `int()` can be called without args apparently so was not a good test case.

```
>>> a.update({b["nope"]: beer()})
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
KeyError: 'nope'
>>> a[b["nope"]] = beer()
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
NameError: name 'beer' is not defined
```

Here you can see the right is evaluated first. I think in your case you are creating a dictionary to pass to `update` and in that case it does evaluate keys first ?